File: js_sdk/src/model/dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';
import {dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules} from './dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules.js';

/**
 * The dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet model module.
 * @module model/dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet
 * @version 0.1.2
 */
export class dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet {
  /**
   * Constructs a new <code>dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet</code>.
   * @alias module:model/dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet
   * @class
   * @param operator {String} Required when exclusion_rule_set is passed. Operator between the exclusion rules within the exclusion rule set. Enum value: OR.
   * @param rules {Array.<module:model/dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules>} Required when exclusion_rule_set is passed. A list of exclusion rules within the exclusion rule set.
   */
  constructor(operator, rules) {
    this.operator = operator;
    this.rules = rules;
  }

  /**
   * Constructs a <code>dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet} obj Optional instance to populate.
   * @return {module:model/dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet} The populated <code>dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet();
      if (data.hasOwnProperty('operator'))
        obj.operator = ApiClient.convertToType(data['operator'], 'String');
      if (data.hasOwnProperty('rules'))
        obj.rules = ApiClient.convertToType(data['rules'], [dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules]);
    }
    return obj;
  }
}

/**
 * Required when exclusion_rule_set is passed. Operator between the exclusion rules within the exclusion rule set. Enum value: OR.
 * @member {String} operator
 */
dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.prototype.operator = undefined;

/**
 * Required when exclusion_rule_set is passed. A list of exclusion rules within the exclusion rule set.
 * @member {Array.<module:model/dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules>} rules
 */
dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.prototype.rules = undefined;

File: java_sdk/src/main/java/business_api_client/DmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import business_api_client.DmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * DmpcustomAudiencerulecreateRuleSpecExclusionRuleSet
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2025-12-02T15:00:15.368932-08:00[America/Los_Angeles]")
public class DmpcustomAudiencerulecreateRuleSpecExclusionRuleSet {
  @JsonProperty("operator")
  private String operator = null;

  @JsonProperty("rules")
  private List<DmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules> rules = new ArrayList<DmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules>();

  public DmpcustomAudiencerulecreateRuleSpecExclusionRuleSet operator(String operator) {
    this.operator = operator;
    return this;
  }

   /**
   * Required when exclusion_rule_set is passed. Operator between the exclusion rules within the exclusion rule set. Enum value: OR.
   * @return operator
  **/
  @Schema(required = true, description = "Required when exclusion_rule_set is passed. Operator between the exclusion rules within the exclusion rule set. Enum value: OR.")
  public String getOperator() {
    return operator;
  }

  public void setOperator(String operator) {
    this.operator = operator;
  }

  public DmpcustomAudiencerulecreateRuleSpecExclusionRuleSet rules(List<DmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules> rules) {
    this.rules = rules;
    return this;
  }

  public DmpcustomAudiencerulecreateRuleSpecExclusionRuleSet addRulesItem(DmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules rulesItem) {
    this.rules.add(rulesItem);
    return this;
  }

   /**
   * Required when exclusion_rule_set is passed. A list of exclusion rules within the exclusion rule set.
   * @return rules
  **/
  @Schema(required = true, description = "Required when exclusion_rule_set is passed. A list of exclusion rules within the exclusion rule set.")
  public List<DmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules> getRules() {
    return rules;
  }

  public void setRules(List<DmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules> rules) {
    this.rules = rules;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DmpcustomAudiencerulecreateRuleSpecExclusionRuleSet dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet = (DmpcustomAudiencerulecreateRuleSpecExclusionRuleSet) o;
    return Objects.equals(this.operator, dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.operator) &&
        Objects.equals(this.rules, dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.rules);
  }

  @Override
  public int hashCode() {
    return Objects.hash(operator, rules);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DmpcustomAudiencerulecreateRuleSpecExclusionRuleSet {\n");
    
    sb.append("    operator: ").append(toIndentedString(operator)).append("\n");
    sb.append("    rules: ").append(toIndentedString(rules)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
